src/js/internal/promisify.ts

```diff
@@ -0,0 +1,79 @@
+const kCustomPromisifiedSymbol = Symbol.for("nodejs.util.promisify.custom");
+const kCustomPromisifyArgsSymbol = Symbol("customPromisifyArgs");
+
+function defineCustomPromisify(target, callback) {
+  Object.defineProperty(target, kCustomPromisifiedSymbol, {
+    value: callback,
+    __proto__: null,
+    configurable: true,
+  });
+
+  return callback;
+}
+
+function defineCustomPromisifyArgs(target, args) {
+  Object.defineProperty(target, kCustomPromisifyArgsSymbol, {
+    __proto__: null,
+    value: args,
+    enumerable: false,
+  });
+  return args;
+}
+
+var promisify = function promisify(original) {
+  if (typeof original !== "function") throw new TypeError('The "original" argument must be of type Function');
+  const custom = original[kCustomPromisifiedSymbol];
+  if (custom) {
+    if (typeof custom !== "function") {
+      throw new TypeError('The "util.promisify.custom" argument must be of type Function');
+    }
+    // ensure that we don't create another promisified function wrapper
+    return defineCustomPromisify(custom, custom);
+  }
+
+  const callbackArgs = original[kCustomPromisifyArgsSymbol];
+
+  function fn(...originalArgs) {
```

Comment:
instead of checking for callbackArgs on every call

this should return a different function which handles the callbackArgs and a function which does not handle them

Reply:
though i see that this was already implemented that way